### ISSUE TITLE
General settings: hidden / disabled items

### ISSUE THREAD

**Issue by hius07** [koreader/koreader#11335]
Starting in https://github.com/koreader/koreader/pull/11327#discussion_r1446495453, invite for broader discussion.

Currently there are 3 coexisting approaches to the general settings availability:

(1) Always available
eg: File browser - Settings - Mosaic and detailed list settings (_available in Classic mode_)
eg: Reader - Settings - Document - Save document (write highlight into PDF) (_available with opened cre document_)

(2) Disabled when inapplicable
eg: Reader - Settings - Taps and gestures - Scrolling (_visible but disabled in page mode_)

(3) Hidden when inapplicable
eg new item: Reader - Settings - Taps and gestures - Long press on text - Auto-scroll when selection reaches a corner (_hidden with opened pdf document_)

My vote is for the (1) option only.
When I am **setting up the environment**, I want to see all the settings and I understand that some of them have effect in certain circumstances only. That is why we call them "general settings".

**Comment by poire-z** [comment]
As detailed in the other discussion, my vote is not for 1) - or at least not for a 1) as a 100% rule.
And I don't think it needs a unique approach (I know you like consistency, but what's best may depend on the context, and I don't think the state we are in with intuitive decisions is bad).

> That is why we call them "general settings".

Who/where do we call them that ? and how do we call other things? If you mean G_reader_settings vs. docsettings, I'm quite sure there are cre stuff that ends up as global things in G_reader_settings.
Aren't stuff in Typography rules general settings ? Should we show them also with PDF?

> When I am **setting up the environment**, I want to see all the settings and I understand that some of them have effect in certain circumstances only. 

"setting up the environment" feels like it's a one-time thing (when you get a new device?). After that, which is happening more often :), you're adjusting stuff to what you actually need at the moment in your specific context.

If there's anything confusing with 2), stuff disabled and not knowing how to get them enabled, a `help_text_when_disabled` acting on tap or long-press to show what's required to enable the feature could be nice - but the visual hint of having a thing disabled and that it's no use taping/going into it at the moment is worth more than any "setting up the environment/show them all" thing (and possibly getting popups sayings "this can't work currently, this only when when x and y").

**Comment by poire-z** [comment]
> (3) Hidden when inapplicable
eg new item: Reader - Settings - Taps and gestures - Long press on text - Auto-scroll when selection reaches a corner (hidden with opened pdf document)

About this specific new stuff not yet merged, and to avoid introducing case (3), would you prefer me to make it (2) ? if ui.paging: disabled & unchecked ?

**Comment by hius07** [comment]
Yes, thanks.

**Comment by offset-torque** [comment]
**"Always available"** option is better for advanced users as you explained here **hius07**:

> When I am setting up the environment, I want to see all the settings and I understand that some of them have effect in certain circumstances only.

I would also prefer to have all the settings available all the time to avoid changing context to save effort.
But this can be confusing for the new/average user. 

Normally users get feedback from their actions while interacting with the UI. This feedback allows them to:
1. Connect the result with their action (I turned on **Reflow** and this PDF became unreadable)
2. And depending on this knowledge, undo the action if necessary (I will turn off **Reflow**)

Losing this cause-effect relationship creates confusion. As **poire-z** said in the other thread:

> there's no need to allow the user to toggle stuff in a context where it doesn't apply and it can't verify what option is best for him.

Even in highly specialized applications like **Blender**, tools and settings are restricted to their contexts because by invoking an out-of-context operation, users can seriously mess up their model and they won't even notice because there won't be any feedback in the current mode.

So in summary, disabling/hiding settings depending on the context is a bit inconvenient for advanced users but much better for normal users.
